### PR TITLE
Document the CC encryption feature

### PR DESCRIPTION
**Pull request by abstratt**
Issue: #23866

### CODE REVIEW

**Comment by abstratt** [comment]
@bot-gradle test BD

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc [hunk 1]
```diff
@@ -204,6 +204,43 @@ If you need to invalidate the cache, simply delete that directory:
 Configuration cache entries are checked periodically (at most every 24 hours) for whether they are still in use.
 They are deleted if they haven't been used for 7 days.
 
+[[config_cache:usage:encryption]]
+=== Encrypting the cache
+
+The configuration cache stores any data set up during the configuration phase.
+As such, if your build configuration includes any sort of sensitive information, you should ensure
```
Comment (hythloda) [inline subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:211]: 🧐 
```suggestion
As such, if your build configuration includes any sensitive information, you should ensure
```

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc [hunk 2]
```diff
@@ -204,6 +204,43 @@ If you need to invalidate the cache, simply delete that directory:
...
+your configuration cache is encrypted.
+
+For Gradle to be able to encrypt the data in the cache, you must provide Gradle with an AES 128-bit key
+via the `GRADLE_ENCRYPTION_KEY` environment variable (see <<#sec:gradle_environment_variables, see Environment variables>>).
+
+==== Generating an encryption key that is compatible with GRADLE_ENCRYPTION_KEY
+
+There are many different ways of producing an encryption key that is compatible with Gradle.
```
Comment (hythloda) [inline subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:219]: 🧐 
```suggestion
There are many ways of producing an encryption key compatible with Gradle.
```

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc [hunk 3]
```diff
@@ -204,6 +204,43 @@ If you need to invalidate the cache, simply delete that directory:
...
+One way is by using a command like this, which should work on Linux and MacOSx, or even on Windows, if using a tool like Cygwin:
+
+```
+echo $(openssl enc -aes-128-cbc -k secret -P | grep key= | sed s/key=// | base64)
+```
+
+You can then use the key produced by that command (which is encoded as a Base64 string) and set it as the value of the
+`GRADLE_ENCRYPTION_KEY` environment variable however you prefer configuring environment variables in your OS.
```
Comment (hythloda) [inline subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:227]: 🧐 
```suggestion
`GRADLE_ENCRYPTION_KEY` environment variable; however, you prefer configuring environment variables in your OS.
```

**Review by hythloda — APPROVED** [review]

File: subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc (hunk 3 above)
Comment (abstratt) [reply subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc:227]: "however" as in "whichever way" - not usual English I gather?

**Comment by abstratt** [comment]
@bot-gradle test BD

**Comment by abstratt** [comment]
New section [here](https://builds.gradle.org/repository/download/Gradle_Release_Check_BuildDistributions/62764186:id/distributions/gradle-8.1-docs.zip!/gradle-8.1-20230307200242%2B0000/docs/userguide/configuration_cache.html#config_cache:usage:encryption)

**Pull request merged**